### PR TITLE
Track `hashchange` event

### LINKED ISSUE #136: Track `hashchange` event
In Chrome, if you set a new text fragment hash programmatically, like
https://github.com/GoogleChromeLabs/text-fragments-polyfill
`location.hash = '#:~:text=module%20of%20util-,functions,-for%20generating%20URLs';`
, the page will be scrolled to the text.

The behavior does not provide such functionality.

### PR DESCRIPTION
**Pull request by Shifu462**
Closes #136

### CODE REVIEW

**Comment by tfmar** [comment]
Hi Mikhail, is the intent here to support highlighting on same-document links, JS-initiated URL changes, etc? If so, [I don't think the spec allows this.](https://wicg.github.io/scroll-to-text-fragment/#:~:text=Forbidden%20on%20a%20same%2Ddocument%20navigation.)

File: src/text-fragments.js [hunk 1]
```diff
@@ -22,19 +22,20 @@ import * as utils from './text-fragment-utils.js';
...
 
-  // Return early when there is no hash.
-  // This is fine, since later added hashes are not relevant for
-  // Text Fragments.
-  if (!hash) {
-    return;
-  }
-
   const init = () => {
+    utils.removeAllMarksInDocument(document);
```
[the remark this answers is not in the document]
Comment (Shifu462) [reply src/text-fragments.js:30]: For some reason thought Chrome removes highlights. Maybe it did back then in April 2022.

Will fix this.

[the remark this answers is not in the document]
Comment (Shifu462) [reply src/text-fragments.js]: Removed this line, removed the `removeAllMarksInDocument` util.

**Comment by bokand** [comment]
Thanks, lgtm - the only thing I'd check is what happens if the hash changes but the same text fragment is still there (i.e. is it ok for the same snippet to get highlighted multiple times)? I'll defer to @tomayac on overall correctness.

**Review by tomayac — APPROVED** [review]
LGTM, thank you.

**Pull request merged**